### PR TITLE
add ceph rbd flexvolume for hyper

### PR DESCRIPTION
**Pull request by bergwolf**
It invokes rbd command to format the disk and saves rbd volume
info for hyper runtime to use.

1. copy built cephrbd binary to flexvolume dir:
`cp out/cephrbd /usr/libexec/kubernetes/kubelet-plugins/volume/exec/hyper~cephrbd/cephrbd`
2. create a rbd volume:
`rbd create -p hyper --size 1G frakti-rbd-vol2`
3. create pod with a `hyper/cephrbd` flexvolume in the yaml (`pool` is optional and default to `hyepr` pool): 
```
$ cat rbd-pod.yaml
apiVersion: v1
kind: Pod
metadata:
  name: nginx-rbd-pod
  labels:
    app: nginx
spec:
  containers:
  - name: nginx
    image: nginx
    imagePullPolicy: IfNotPresent
    ports:
    - containerPort: 80
    volumeMounts:
    - name: nginx-persistent-storage
      mountPath: /var/lib/nginx
  volumes:
  - name: nginx-persistent-storage
    flexVolume:
      driver: "hyper/cephrbd"
      fsType: "ext4"
      options:
        pool: "hyper"
        volumeID: "frakti-rbd-vol1"
```

Results:
```
~/bergwolf$ sudo kubectl get pods
NAME                      READY     STATUS             RESTARTS   AGE
alpine                    0/1       Succeeded          0          1d
alpine-6fd4d479cd-5mdcl   0/1       CrashLoopBackOff   25         12h
~/bergwolf$ sudo kubectl create -f rbd-pod.yaml
pod "nginx-rbd-pod" created
~/bergwolf$ sudo kubectl get pods
NAME                      READY     STATUS             RESTARTS   AGE
alpine                    0/1       Succeeded          0          1d
alpine-6fd4d479cd-5mdcl   0/1       CrashLoopBackOff   26         12h
nginx-rbd-pod             1/1       Running            0          6s
~/bergwolf$ sudo kubectl exec nginx-rbd-pod df
Filesystem     1K-blocks     Used Available Use% Mounted on
share_dir       80603936 50201268  26285164  66% /
devtmpfs           58664        0     58664   0% /dev
tmpfs              60864        0     60864   0% /dev/shm
rootfs             58664    12748     45916  22% /etc/hostname
/dev/sda          999320     1288    929220   1% /var/lib/nginx
```

### CODE REVIEW

File: pkg/flexvolume/rbd/flexvolume.go [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+}
+
+type driverOp func(*FlexVolumeDriver, []string) (map[string]interface{}, error)
+
+type cmdInfo struct {
+	numArgs int
+	run     driverOp
+}
+
+var commands = map[string]cmdInfo{
```
Comment (resouer) [inline pkg/flexvolume/rbd/flexvolume.go]: Is it possible to share this part of common code below?

File: pkg/flexvolume/rbd/util.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+	return &rbdDevice{
+		rbdBinPath: rbdPath,
+		volId:      volId,
+		pool:       pool,
+		fstype:     fstype,
+	}, nil
+}
+
+func (r *rbdDevice) mapDevice() (string, error) {
+	glog.Infof("map device %s", r.volId)
```
Comment (resouer) [inline pkg/flexvolume/rbd/util.go]: `Infof()` is too much here, how about V(3) or V(5)?

File: pkg/flexvolume/rbd/util.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+		return r.mappedDev, nil
+	}
+
+	mappedDeviceByte, err := exec.Command(r.rbdBinPath, "map", r.volId, "-p", r.pool).CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("fail to map rbd: %v", err)
+	}
+
+	r.mappedDev = strings.TrimSpace(string(mappedDeviceByte))
+	glog.Infof("volume %s mapped as %s", r.volId, r.mappedDev)
```
Comment (resouer) [inline pkg/flexvolume/rbd/util.go]: Same here and other places :D

File: pkg/flexvolume/rbd/flexvolume.go (hunk 1 above)
Comment (bergwolf) [reply pkg/flexvolume/rbd/flexvolume.go]: Done.

File: pkg/flexvolume/rbd/util.go (hunk 2 above)
Comment (bergwolf) [reply pkg/flexvolume/rbd/util.go]: Fixed all of them.

**Pull request merged**